test/log.ts

```diff
@@ -356,46 +383,41 @@ describe('Log', () => {
         resource: CUSTOM_RESOURCE,
       }) as WriteOptions;
 
-      log.logging.loggingService.writeLogEntries = (
-        reqOpts: {},
-        gaxOpts: {}
-      ) => {
-        assert.deepStrictEqual(reqOpts, {
-          logName: log.formattedName_,
-          entries: [ENTRY],
-          resource: CUSTOM_RESOURCE,
-        });
-
-        assert.strictEqual(gaxOpts, undefined);
-      };
-
       await log.write(ENTRY, optionsWithResource);
+      assert(
+        writeLogStub.calledOnceWithExactly(
+          {
+            logName: log.formattedName_,
+            entries: [ENTRY],
+            resource: CUSTOM_RESOURCE,
+          },
+          undefined
+        )
+      );
     });
 
     it('should cache a detected resource', async () => {
       const fakeResource = 'test-level-fake-resource';
-      fakeMetadata.getDefaultResource = async () => {
-        return fakeResource;
-      };
-      log.logging.loggingService.writeLogEntries = () => {
-        assert.strictEqual(log.logging.detectedResource, fakeResource);
-      };
+      fakeResourceStub.resolves(fakeResource);
+
       await log.write(ENTRY);
+      assert(writeLogStub.calledOnce);
+      assert.strictEqual(log.logging.detectedResource, fakeResource);
     });
 
     it('should re-use detected resource', async () => {
-      log.logging.detectedResource = 'environment-default-resource';
-      fakeMetadata.getDefaultResource = () => {
-        throw new Error('Cached resource was not used.');
-      };
-      // tslint:disable-next-line no-any
-      log.logging.loggingService.writeLogEntries = (
-        // tslint:disable-next-line no-any
-        reqOpts: any
-      ) => {
-        assert.strictEqual(reqOpts.resource, log.logging.detectedResource);
-      };
+      const reusableDetectedResource = 'environment-default-resource';
```

Comment:
Appreciate the split out to its own variable to be used in the test, rather than the instance property.